src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -83,6 +103,10 @@ def run(self):
 
         def on_assign(consumer, partitions):
             for partition in partitions:
+                if self.resolve_partition_force_offset:
+                    partition = self.resolve_partition_force_offset(partition)
+                    self.consumer.assign([partition])
```

Comment:
Did you test this with multiple partitions or copy this pattern from somewhere that we've tested it with multiple partitions? I thought calling `assign` replaced the previous assignment in its entirety — i.e. `assign([a])` and `assign([b])` would result in the assignment state being `[b]`, not `[a, b]`.

Reply:
I didn't test this... I was considering just doing a bulk `assign` with all the partitions but I assumed it was a minor efficiency improvement, so didn't do it. It looks like we're doing what I suggested here: https://github.com/getsentry/sentry/blob/f952fedefe679d632077a2b334a913e9e68cfd91/src/sentry/eventstream/kafka/consumer.py#L297, so I'll just follow that pattern instead.